### PR TITLE
Extra works for application deployment

### PR DESCRIPTION
**Pull request by stndn**
Here are some changes to incorporate additional package (virtualenv) as well as setup of the application.

The application setup includes creation of new linux user, checkout code from gitlab, install python packages, start/restart gunicorn.

Still missing are the proper gunicorn config file, the setup of database needed for application, and the config for nginx to point to the gunicorn instance.

### CODE REVIEW

File: etiqa-flask-playbook/tasks/dependencies.yml [hunk 1]
```diff
@@ -3,14 +3,15 @@
...
   - name: Install git
     package:
       name: git
-      state: present 
+      state: present
+
+  - name: Install virtualenv
+    package:
+      name: python3-virtualenv
+      state: present
```
Comment (lowks) [inline etiqa-flask-playbook/tasks/dependencies.yml:16]: Bro this one right you can summarise this to a one line using `with_items` Something along the lines of this: https://ansible-tips-and-tricks.readthedocs.io/en/latest/os-dependent-tasks/installing_packages/

File: etiqa-flask-playbook/tasks/checkout_code.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+- name: Read git credentials
+  include_vars:
+    file: ../vars/gitlab.yml
+  no_log: true
```
Comment (lowks) [inline etiqa-flask-playbook/tasks/checkout_code.yml:5]: Why do you need this ?

File: etiqa-flask-playbook/tasks/checkout_code.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+     repo: "https://{{ gituser }}:{{ gitpass }}@gitlab.com/ata-team/ata-pillar-v/social-media-search-api-official.git"
+     dest: "{{ smas.app_path }}"
+
+- name: Change the ownership of the directory
+  become: yes
+  file:
+    path: "{{ smas.app_path }}"
+    owner: "{{ smas.username }}"
+    recurse: yes
+  register: chgrpout
```
Comment (lowks) [inline etiqa-flask-playbook/tasks/checkout_code.yml:19]: And why do you register this ? Where are you planning to use this ?

File: etiqa-flask-playbook/tasks/install_flask.yml [hunk 4]
```diff
@@ -4,9 +4,68 @@
...
+      environment:
+        twitter_consumer_key: "{{ twitter.twitter_consumer_key }}"
+        twitter_consumer_secret: "{{ twitter.twitter_consumer_secret }}"
+        twitter_access_token_key: "{{ twitter.twitter_access_token_key }}"
+        twitter_access_token_secret: "{{ twitter.twitter_access_token_secret }}"
+
+
+    # TODO:
+    # - config file for gunicorn to specify log file location, port number, number of workers, timeout, etc
+    # - Create mysql user, db, table, all for the SMAS app. SQL file is at /home/smas/smas-app/setup/setup.sql
```
Comment (lowks) [inline etiqa-flask-playbook/tasks/install_flask.yml:65]: Have a look at this via var definition for `geerlingguy.mysql` role. You can do it via that.

File: etiqa-flask-playbook/tasks/checkout_code.yml (hunk 2 above)
Comment (stndn) [reply etiqa-flask-playbook/tasks/checkout_code.yml:5]: I'm storing the credentials to login to gitlab, which will be used to checkout the code from gitlab. You can refer to line 10. Based on suggestion from https://www.middlewareinventory.com/blog/ansible-git-example/

File: etiqa-flask-playbook/tasks/checkout_code.yml (hunk 3 above)
Comment (stndn) [reply etiqa-flask-playbook/tasks/checkout_code.yml:19]: We will serve the application using separate account without root privilege. This is based on the guide here: https://www.digitalocean.com/community/tutorials/how-to-serve-flask-applications-with-gunicorn-and-nginx-on-ubuntu-18-04

Comment (stndn) [reply etiqa-flask-playbook/tasks/checkout_code.yml:19]: > And why do you register this ? Where are you planning to use this ?

Oh, now I get what you mean. the `register` on line 19.
You are correct that it is not needed. Copy/paste mistake. Will remove it.

**Pull request merged**